### PR TITLE
feat: Network switching strategy adjustment

### LINKED ISSUE #254: Network switching strategy is too aggressive
### The problem
Currently, whenever the current network changes from a supported network to an unsupported network, godwoken-bridge requests to switch the current network to a supported one, which can be too aggressive as it can be triggered even if the user was focusing on other apps.

Maybe when we detect a network change, we should notify and ask the user to switch the network first, instead of requesting the switch directly. An example of the process:
1. Detect if the current network changes to an unsupported network
2. If it does, ask to switch the network via confirm dialog
3. If confirmed, request to switch the network

### Contributors
Thanks to @gpBlockchain @Dawn-githup for pointing out the issue.

### PR DESCRIPTION
**Pull request by GitOfJason**
### Describe
This pull request optimizes the network switching policy.
When network changes are detected,Not directly switch the network,Instead, the user is notified and asked to switch the network

### Related issues
- fix #254

### CODE REVIEW

**Comment by Flouse** [comment]
test link: https://godwoken-bridge-mainnet-7jeqgbmsp-godwoken.vercel.app

Please check the following test cases:
- [ ] switch from Godwoken mainnet_v0 to mainnet_v1 
- [ ] switch from Godwoken mainnet_v1 to mainnet_v0

cc @Dawn-githup

**Review by ShookLyngs — CHANGES_REQUESTED** [review]

**Comment by Dawn-githup** [comment]
Click the blank area outside the pop-up box to prompt that the pop-up box will also disappear 

Misclick may occur. I think it should not disappear unless the user closes it or clicks cancel confirm

File: apps/godwoken-bridge/src/views/GodwokenBridge.tsx [hunk 1]
```diff
@@ -8,32 +8,64 @@ import { addNetwork } from "../utils/addNetwork";
...
-      <Outlet />
-      <PageFooter />
-    </Page>
+    <>
+      <Page>
+        <PageHeader />
+        <Outlet />
+        <PageFooter />
+      </Page>
+      <NetworkMismatchModal
```
[the remark this answers is not in the document]
Comment (GitOfJason) [reply apps/godwoken-bridge/src/views/GodwokenBridge.tsx:63]: That's a good idea

**Comment by Dawn-githup** [comment]
> test link: https://godwoken-bridge-mainnet-7jeqgbmsp-godwoken.vercel.app
> 
> Please check the following test cases:
> 
> * [ ]  switch from Godwoken mainnet_v0 to mainnet_v1
> * [ ]  switch from Godwoken mainnet_v1 to mainnet_v0
> 
> cc @Dawn-githup
> 
> ## Reference
> Godwoken Public Networks - https://docs.godwoken.io/connectionInfo

regression test case
- [ ] switch from Godwoken mainnet to testnet
- [ ] switch from Godwoken testnet to mainnet
- [ ] switch from Godwoken mainnet&testnet to ethernet& Other networks
- [ ] Compatible display of the pop-up box on the mobile terminal

**Comment by GitOfJason** [comment]
> Click the blank area outside the pop-up box to prompt that the pop-up box will also disappear
> 
> Misclick may occur. I think it should not disappear unless the user closes it or clicks cancel confirm

Fixed

**Comment by Dawn-githup** [comment]
> > Click the blank area outside the pop-up box to prompt that the pop-up box will also disappear
> > Misclick may occur. I think it should not disappear unless the user closes it or clicks cancel confirm
> 
> Fixed

Thanks @GitOfJason

**Review by Dawn-githup — APPROVED** [review]

**Review by ShookLyngs — APPROVED** [review]

**Comment by Flouse** [comment]
When I visited https://godwoken-bridge-mainnet-qkjgqqcvp-godwoken.vercel.app/#/v0 , and switch between mainnet_v1 and mainnet_v0, it showed

image

@ShookLyngs @Dawn-githup could you confirm this issue?

**Comment by GitOfJason** [comment]
> 

@Flouse I know what the problem is and I will fix it

**Comment by GitOfJason** [comment]
> @Flouse The network switching part only detects if the network from the provider changes, and our app does not actually need to add mainnet/testnet v0 networks to the provider. Which means users add v1 networks to their providers even if they're using the v0 bridge. However, it does look weird.
> 
> @GitOfJason How do you plan to fix it?

Sorry, I understand it wrong.  As @ShookLyngs  said, in order to make the network name look less strange, we should specify a display name instead of using layer2Config. CHAIN_ NAME as the displayed name

**Comment by ShookLyngs** [comment]
> Sorry, I understand it wrong. As @ShookLyngs said, in order to make the network name look less strange, we should specify a display name instead of using `layer2Config. CHAIN_ NAME` as the displayed name

It seems your first attemp was friendly enough, how about using the `isMainnet` value:
```typescript
const networkName = `Godwoken ${isMainnet ? "Mainnet" : "Testnet"}`;
```

**Comment by GitOfJason** [comment]
> > Sorry, I understand it wrong. As @ShookLyngs said, in order to make the network name look less strange, we should specify a display name instead of using `layer2Config. CHAIN_ NAME` as the displayed name
> 
> It seems your first attemp was friendly enough, how about using the predefined `isMainnet` variable:
> 
> ```ts
> const networkName = `Godwoken ${isMainnet ? "Mainnet" : "Testnet"}`;
> ```

@Flouse @ShookLyngs  I will fix it in this way

**Pull request merged**